File: Algorithm And Data Structure/Linked List/insertion.c
#include <iostream>
#include <bits/stdc++.h>

using namespace std;

struct Node
{
    int key;
    Node *next;
    
    Node(int x)
    {
        key = x;
        next = NULL;
    }
};

Node *insertNode(Node *head, int key)
{
    Node *temp = new Node(key);
    temp->next = head;
    return temp;
}

Node *insertEnd(Node *head, int key)
{
    if(head == NULL)
    return  new Node(key);
    
    Node *curr = head;
    while(curr->next != NULL)
        curr = curr->next;
    curr->next = new Node(key);
    return head;
}
void printList(Node *head)
{
    Node *curr = head;
    
    while(curr != NULL)
    {
        cout<<curr->key<<" ";
        curr= curr->next;
    }
}

int main() {
	Node *head = NULL;
	head = insertNode(head,7);
	head = insertNode(head,18);
	head = insertNode(head,27);
	head = insertNode(head,99);

	head = insertEnd(head, 7);
	head = insertEnd(head,18);
	head = insertEnd(head,27);
	head = insertEnd(head,99);
	printList(head);
	return 0;
}

File: Algorithm And Data Structure/Linked List/insertion.java
/*package whatever //do not write package name here */

import java.io.*;
import java.util.*;

class Node
{
    int key;
    Node next;
    
    Node(int x)
    {
        key = x;
    }
    
};

class LList{
    Node head;
    
    void insertNode(int x)
    {
    Node temp = new Node(x);
    temp.next = head;
    head = temp;
    }
    void insertEnd(int x)
    {
        Node temp = new Node(x);
        if(head == null)
        {
            head = temp;
            return;
        }
        Node curr = head;
        while(curr.next != null)
        {
            curr = curr.next;
        }
        curr.next = temp;
        
    }
    void printNode()
    {
        Node curr = head;
        
        while(curr != null)
        {
            System.out.print(curr.key+ " ");
            curr = curr.next;
        }
    }
    
}

class GFG {
	public static void main (String[] args) {
	    
	    LList l = new LList();
	    l.insertNode(7);
	    l.insertNode(77);
	    l.insertNode(99);
	    l.insertEnd(7);
	    l.insertEnd(77);
	    l.insertEnd(99);
	    
		l.printNode();
	}
}